### PR TITLE
Documentation - update dead/redirected links

### PR DESCRIPTION
**Pull request by lb-**
- Update a bunch of links that now redirect to new URLs
- Replace a few links that no longer work with their current equivalent
- Update some links for the documentation README.md file to align with the new MyST syntax usage
- When linking to external dependency documentation, use the `stable` link not the `latest` link as that is more appropriate

### CODE REVIEW

File: docs/advanced_topics/deploying.md [hunk 1]
```diff
@@ -7,13 +7,13 @@ See the section on [performance](performance) for the non-Python services we rec
...
-[Divio Cloud](https://divio.com/) is a Dockerised cloud hosting platform for Python/Django that allows you to launch and deploy Wagtail projects in minutes.
+[Divio Cloud](https://www.divio.com/) is a Dockerised cloud hosting platform for Python/Django that allows you to launch and deploy Wagtail projects in minutes.
 With a free account, you can create a Wagtail project. Choose from a:
 
--   [site based on the Wagtail Bakery project](https://divio.com/wagtail), or
+-   [site based on the Wagtail Bakery project](https://www.divio.com/wagtail/), or
 -   [brand new Wagtail project](https://control.divio.com/control/project/create) (see the [how to get started notes](https://docs.divio.com/en/latest/introduction/wagtail/)).
 
-Divio Cloud also hosts a [live Wagtail Bakery demo](https://divio.com/wagtail) (no account required).
+Divio Cloud also hosts a [live Wagtail Bakery demo](https://divio.com/wagtailhttps://www.divio.com/wagtail/) (no account required).
```
Comment (zerolab) [inline docs/advanced_topics/deploying.md:16]:
```suggestion
Divio Cloud also hosts a [live Wagtail Bakery demo](https://www.divio.com/wagtail/) (no account required).
```

File: docs/contributing/security.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ On the day of disclosure, we will take the following steps:
    The commit messages for these patches will indicate that they are for security issues, but will not describe the issue in any detail; instead, they will warn of upcoming disclosure.
 2. Issue the relevant release(s), by placing new packages on [the Python Package Index](https://pypi.org/project/wagtail/), tagging the new release(s) in Wagtail\'s GitHub repository and updating Wagtail\'s [release notes](../releases/index).
 3. Post a public entry on [Wagtail\'s blog](https://wagtail.org/blog/), describing the issue and its resolution in detail, pointing to the relevant patches and new releases, and crediting the reporter of the issue (if the reporter wishes to be publicly identified).
-4. Post a notice to the [Wagtail support forum](https://groups.google.com/d/forum/wagtail) and Twitter feed ([\@WagtailCMS](https://twitter.com/wagtailcms)) that links to the blog post.
+4. Post a notice to the [Wagtail support forum](https://groups.google.com/g/wagtail) and Twitter feed ([\@WagtailCMS](https://twitter.com/wagtailcms)) that links to the blog post.
```
Comment (zerolab) [inline docs/contributing/security.md:44]: question: are we still actively encouraging people to use Google Groups? My impression was we switched to GitHub Discussions

File: docs/support.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ If you have any problems or questions about working with Wagtail, you are invite
 
 ## Mailing list
 
-For topics and discussions that do not fit Stack Overflow's question-and-answer format, there is a Wagtail Support mailing list at [groups.google.com/d/forum/wagtail](https://groups.google.com/d/forum/wagtail).
+For topics and discussions that do not fit Stack Overflow's question-and-answer format, there is a Wagtail Support mailing list at [groups.google.com/d/forum/wagtail](https://groups.google.com/g/wagtail).
```
Comment (zerolab) [inline docs/support.md:13]: suggestion: let's link to GitHub Discussions

**Review by zerolab — CHANGES_REQUESTED** [review]
Had a quick scan and made a couple of suggestions

File: docs/advanced_topics/deploying.md (hunk 1 above)
Comment (lb-) [reply docs/advanced_topics/deploying.md:16]: Thanks - must have been a bad find/replace.

File: docs/contributing/security.md (hunk 2 above)
Comment (lb-) [reply docs/contributing/security.md:44]: I thought you would ask this haha. I thought the same but it looks like Matt is still posting to it.

Screen Shot 2022-10-04 at 8 48 01 pm

I think we should update the link but probably a good question for the core team - I feel like we have discussed this before but cannot remember the outcome.

File: docs/support.md (hunk 3 above)
Comment (lb-) [reply docs/support.md:13]: yep - will update this one, to align with the readme.

Comment (lb-) [reply docs/support.md:13]: Actually we refer to both slack and the GitHub discussions on this same page further down.

I might leave this as is then until we get a call on whether we should remove references to the Google Group completely.

**Review by zerolab — APPROVED** [review]
LGTM.

Let's follow up the Google Groups vs GitHub Discussion bit in Slack

**Comment by lb-** [comment]
Thanks - will rebase, add changelog and merge in

**Pull request merged**